### PR TITLE
Docs/hdk ergonomics

### PR DESCRIPTION
**Pull request by jost-s**
### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [x] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/hdi/src/lib.rs [hunk 1]
```diff
@@ -50,6 +50,24 @@
 //! values can be validated. All of these validation rules are written in a central callback
 //! which is called by the Holochain engine for each operation.
 //!
+//! There's a helper type called [`OpType`](holochain_integrity_types::OpType) available for easy
+//! access to all link and entry variants when validating an operation.
```
Comment (maackle) [inline crates/hdi/src/lib.rs:54]:
```suggestion
//! There's a helper type called [`OpType`](holochain_integrity_types::OpType) available for easy
//! access to all link and entry variants when validating an operation. In many cases, this type can 
//! be easier to work with than the bare [`Op`](holochain_integrity_types::Op), which contains the
//! same information as `OpType`, but the former has a flatter data structure, whereas the latter has
//! a more deeply nested structure.
```

Comment (maackle) [reply crates/hdi/src/lib.rs:54]: Also is this in accord with [our rustdoc guide](https://hackmd.io/PeLqwcSKRSOtl0X0BJIIJQ#Using-Markdown) which recommends "reference-style" links?

**Review by maackle — APPROVED** [review]
Looks good

File: crates/hdi/src/lib.rs (hunk 1 above)
Comment (jost-s) [reply crates/hdi/src/lib.rs:54]: @maackle Thanks for bringing this up about the `reference-style` links. How I see it, I'd generally use in-line references and only use reference-style links when there are multiple occurrences of the same type. I've changed that in the HackMD, open for discussion there!

**Pull request merged**